var/spack/repos/builtin/packages/cabana/package.py

```diff
@@ -67,6 +68,13 @@ def cmake_args(self):
             '-DBUILD_SHARED_LIBS=%s' % (
                 'On' if '+shared'  in self.spec else 'Off')
         ]
+
+        # Enable Cabana submodules based on flags above
```

Comment:
These don't hurt, but I am wondering if they are actually needed...

Isn't CMake detecting them automatically?

Reply:
Actually the options are called `Cabana_REQUIRE_<foo>`, the `ENABLE` options are internal and set to whether the package was found automatically:
https://github.com/ECP-copa/Cabana/blob/master/CMakeLists.txt#L84

Reply:
It wasn't for me, no. Cabana cmake doesn't look for hypre and then conditionally enable hypre and heffte support. As a result, Cajita_Config.hpp doesn't include the hypre include file or hypre support if this isn't enabled.

Reply:
Hmmm. Let me try again, then. If I just added +hypre to the spec for my spack environment, concretized it, and then did an install, Cabana still didn't have HYPRE support enabled. I'll see if I can track it down better.

Reply:
@streeve do you know?

Reply:
I was able to reproduce the issue. Adding the `Cabana_REQUIRE_` flags below will give clear error messages. It looks like there's an issue with the spack required heffte/hypre versions and the actual CMake required versions

Reply:
I think we need to depend on `hypre-cmake+shared`.